pkgs/applications/networking/msmtp/paths.patch

```diff
@@ -1,45 +1,43 @@
 diff --git a/scripts/msmtpq/msmtpq b/scripts/msmtpq/msmtpq
-index 1b39fc6..4baa19b 100755
+index d8b4039..2bf02e5 100755
 --- a/scripts/msmtpq/msmtpq
 +++ b/scripts/msmtpq/msmtpq
-@@ -70,8 +70,8 @@ MSMTP=msmtp
+@@ -71,7 +71,7 @@ fi
  ##            ( chmod 0700 msmtp.queue )
  ##
- ## the queue dir - modify this to reflect where you'd like it to be  (no quotes !!)
--Q=~/.msmtp.queue
--[ -d "$Q" ] || mkdir -m 0700 "$Q" || \
+ ## the queue dir - export this variable to reflect where you'd like it to be  (no quotes !!)
+-Q=${Q:-~/.msmtp.queue}
 +Q=${MSMTP_QUEUE:-~/.msmtp.queue}
-+[ -d "$Q" ] || mkdir -m 0700 -p "$Q" || \
+ [ -d "$Q" ] || mkdir -m 0700 -p "$Q" || \
    err '' "msmtpq : can't find or create msmtp queue directory [ $Q ]" ''     # if not present - complain ; quit
  ##
- ## set the queue log file var to the location of the msmtp queue log file
-@@ -84,7 +84,10 @@ Q=~/.msmtp.queue
- ##     (doing so would be inadvisable under most conditions, however)
+@@ -85,8 +85,10 @@ Q=${Q:-~/.msmtp.queue}
  ##
- ## the queue log file - modify (or comment out) to taste  (but no quotes !!)
--LOG=~/log/msmtp.queue.log
+ ## the queue log file - export this variable to change where logs are stored  (but no quotes !!)
+ ##                      Set it to "" (empty string) to disable logging.
+-[ -v LOG ] || LOG=~/log/msmtp.queue.log
 +LOG=${MSMTP_LOG:-~/log/msmtp.queue.log}
-+[ -d "$(dirname "$LOG")" ] || mkdir -p "$(dirname "$LOG")"
+ [ -d "$(dirname "$LOG")" ] || mkdir -p "$(dirname "$LOG")"
 +
 +JOURNAL=@journal@
  ## ======================================================================================
  
  ## msmtpq can use the following environment variables :
-@@ -138,6 +141,7 @@ on_exit() {                          # unlock the queue on exit if the lock was
+@@ -139,6 +141,7 @@ on_exit() {                          # unlock the queue on exit if the lock was
  ## display msg to user, as well
  ##
  log() {
 +  local NAME=msmtpq
    local ARG RC PFX
    PFX="$('date' +'%Y %d %b %H:%M:%S')"
                                       # time stamp prefix - "2008 13 Mar 03:59:45 "
-@@ -155,10 +159,19 @@ log() {
+@@ -156,10 +159,19 @@ log() {
      done
    fi
  
 +  if [ "$JOURNAL" == "Y" ]; then
-+    for ARG ; do
-+      [ -n "$ARG" ] && \
++    for ARG; do
++      [ -n "$ARG" ] &&
 +        echo "$ARG" | systemd-cat -t $NAME -p info
```

Comment:
I would quote `$NAME` here. It’s not necessary since we know that `NAME` won’t contain spaces, but there’s a small chance that it could contain spaces in the future. (At the moment, I’m biased towards always quoting variables because [this issue](https://github.com/marlam/msmtp-mirror/issues/87) was annoying to deal with and annoying to fix).